cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java

```diff
@@ -509,6 +511,7 @@ protected Future<ConnectorStatusAndConditions> maybeCreateOrUpdateConnector(Reco
                         .compose(conditions ->
                             apiClient.statusWithBackOff(reconciliation, new BackOff(200L, 2, 10), host, port, connectorName)
                                 .compose(createConnectorStatusAndConditions(conditions)))
+                        .compose(status -> autoRestartFailedConnectorAndTasks(reconciliation, host, apiClient, connectorName, connectorSpec, status, resource))
```

Comment:
Is this the right place in the flow to restart failed connectors and tasks? Since we call createConnectorStatusAndConditions on the previous line will this mean that after we have restarted any failing connectors or tasks the connector status will still show it as failed? Do we need a new call to get the status?

Reply:
I set the auto restart after the api status call to get failed connectors and tasks. As is, failed connectors and tasks will stay in failed state until next reconciliation even if the restart was successful. I don't know if I should add an another call to fetch connectors and tasks statuses after auto restarting or it is OK to wait until next reconciliation to have the status.

Reply:
Yes, I forgot about `RESTARTING`, that would be a much better fit.

Reply:
Yeah that works well, better than waiting around to see if it restarted. It's then clear to the user that the operator has taken an action. So we just need to make sure at the end of the reconcile loop that the status we got back is included in the status update. That might mean you still need to have this call higher up.